File: Examenes/ExamenPSP/2DAM_PSP-main/1 - Llamada API Asincrona/TP 1 - Llamada API Asincrona/src/main/java/ui/pantallas/common/BasePantallaController.java
package ui.pantallas.common;

import ui.pantallas.principal.PrincipalController;


public class BasePantallaController {

    private PrincipalController principalController;

    public PrincipalController getPrincipalController() {
        return principalController;
    }

    public void setPrincipalController(PrincipalController principalController) {
        this.principalController = principalController;
    }

    public void principalCargado() {
        //metodo que utilizaran los controllers para inicializar cosas con el principal controller
    }
}

File: PSP/arreglitos/src/main/kotlin/ui/pantallas/common/BasePantallaController.kt
package ui.pantallas.common

import ui.pantallas.mainscreen.MainController

open class BasePantallaController {

    private var principalController: MainController? = null

    fun getPrincipalController(): MainController? {
        return principalController
    }

    fun setPrincipalController(principalController: MainController?) {
        this.principalController = principalController
    }

    open fun principalCargado() {
        // principalCargado() is called when the screen is loaded
    }
}
